modules/lang-groovy/src/test/java/org/elasticsearch/messy/tests/SearchFieldsTests.java

```diff
@@ -367,6 +367,28 @@ public void testScriptFieldUsingSource() throws Exception {
         assertThat(((Map<?, ?>) sObj2Arr3.get(0)).get("arr3_field1").toString(), equalTo("arr3_value1"));
     }
 
+    public void testScriptFieldsForNullReturn() throws Exception {
+        createIndex("test");
+        client().admin().cluster().prepareHealth().setWaitForEvents(Priority.LANGUID).setWaitForYellowStatus().execute().actionGet();
+
+        client().prepareIndex("test", "type1", "1")
+            .setSource(jsonBuilder().startObject().startObject("obj1").field("test", "something").endObject())
```

Comment:
Since you don't care about the body of the source maybe use something like `setSource("foo", "bar")`.